Datahub: Some more fixes, style and test improvements

tkohr opened the pull request:
PR includes the following improvements:

Refactoring and fixes:
- moves section titles to components and hides "Downloads" if no formats can be retrieved from WFS like http://localhost:4200/dataset/urn:isogeo:metadata:uuid:1355be63-7f12-41f7-bcc8-724146679eb3
- reset feature info selection when map context changes
- improve tests for `data-downloads.component` to prevent false positives

Styling improvements:
- add background and border to map for rendering
- prevent cut shadows for API cards and customize horizontal scrollbar
- make related records card (click) more intuitive
- add spinning loader for search results

jahow reviewed (approved):
Tested, improvements are working but I'm not a big fan of the custom scrollbar. Was that a requirement? I think we generally avoid these kind of customizations as they only work in certain browsers (firefox in that case if I'm not mistaken) and might degrade user experience.

Thanks!

tkohr commented on translations/en.json:
```diff
@@ -103,24 +103,23 @@
   "record.action.download": "Download",
   "record.action.view": "View",
+  "record.metadata.about": "About",
+  "record.metadata.access": "Access",
+  "record.metadata.api": "API",
   "record.metadata.contact": "Contact",
-  "record.metadata.description": "Description",
   "record.metadata.details": "Details",
+  "record.metadata.download": "Downloads",
   "record.metadata.keywords": "Keywords",
+  "record.metadata.links": "Links",
   "record.metadata.origin": "Lineage",
+  "record.metadata.preview": "Preview",
+  "record.metadata.related": "Related records",
```
ok, I'll prevent touching on this in the future

tkohr commented:
Thanks for the review @jahow. I got rid of the customized scrollbar and moved the spinning loader to its own component. Also added a value to the `BehaviourSubject` which caused the rebase.
